mongo/client.go

```diff
@@ -307,6 +310,11 @@ func (c *Client) Disconnect(ctx context.Context) error {
 	if disconnector, ok := c.deployment.(driver.Disconnector); ok {
 		return replaceErrors(disconnector.Disconnect(ctx))
 	}
+
+	if c.httpClient == http.DefaultClient {
+		c.httpClient.CloseIdleConnections()
+	}
```

Comment:
Calling `CloseIdleConnections` on the package-global `http.DefaultClient` seems like it could interact unexpectedly with other code that may also use `http.DefaultClient`. We can resolve that concern by always using an `http.Client` instance only used by the `mongo.Client`. Is there a reason to use `http.DefaultClient`?